README.md

```diff
@@ -32,24 +56,234 @@ want the tool to work correctly.
+```
+
+where `<component name>` is the name of component/target you want to be able to load.
+While the component is optional, this is recommended to make sure the correct component is loaded.
+
+Since you rarely only want to load a single component in a `stack` project, we have special syntax to be able to conveniently specify which directory belongs to which component. It is basically a [multi-cradle](#multi-cradle).
+
+```yaml
+cradle:
+  stack:
+    - path: "./src"
+      component: "hie-bios:lib"
+    - path: "./exe"
+      component: "hie-bios:exe:hie-bios"
+    - path: "./tests/BiosTests.hs"
+      component: "hie-bios:test:hie-bios"
+    - path: "./tests/ParserTests.hs"
+      component: "hie-bios:test:parser-tests"
+```
+
+Here you can see two important features:
+  * We provide a mapping from a filepath to component.
+    * That way, we specify that a file such as `./src/HIE/Bios.hs` belongs to the component `hie-bios:lib`.
+  * The filepath can be a file.
+    * This is convenient if components are overlapping.
+
+#### Debugging a `stack` cradle
+
+If you find that `hie-bios` can't load a certain component or file, you may run `stack repl` and `stack repl <component name>` to see if `stack` succeeds building your project. Chances are that there is a problem in your project and if you fix that, `hie-bios` will succeed to load it. Otherwise, please open an [issue](https://github.com/mpickering/hie-bios/issues/new).
+
+#### Ignoring directories
+
+You can combine the [multi-cradle](#multi-cradle) with a [none-cradle](#none-cradle) to ignore all source files in a certain directory. The syntax is a bit verbose:
+
+```yaml
+cradle:
+  multi:
+    - path: "./tests/projects"
+      config:
+        cradle:
+          none:
+    - path: "./"
+      config:
+        cradle:
+          stack:
+            - path: "./src"
+              component: "hie-bios:lib"
+            - path: "./exe"
+              component: "hie-bios:exe:hie-bios"
+            - path: "./tests/BiosTests.hs"
+              component: "hie-bios:test:hie-bios"
+            - path: "./tests/ParserTests.hs"
+              component: "hie-bios:test:parser-tests"
+```
+
+This way, we specify that we do not want to load any files in our test project directories.
+
+#### Internal Libraries
+
+Internal libraries are not well supported in `stack`. Since the syntax [`stack repl <internal library name>`](https://github.com/commercialhaskell/stack/issues/4564) doesn't work, `hie-bios` will generally not work with internal libraries using `stack`.
+
+### Cabal
+
+To use `cabal`, the basic explicit configuration looks similar.
+
+```yaml
+cradle:
+  cabal:
+```
+
+The implication of this configuration is a bit different, though. Given a source file to load, we will use `cabal repl <filename>` to find the component of the given filepath.
+
+Due to an unfortunate [bug](https://github.com/haskell/cabal/issues/6622), this fails on some files with `cabal` versions older than `3.4`.
```

Comment:
```suggestion
This configuration should work in (almost) every standard project setup, since `cabal` finds the component associated to a given source file.
However, due to an unfortunate [bug](https://github.com/haskell/cabal/issues/6622), this fails on some files if the `cabal` version is older than `3.4`. 
```